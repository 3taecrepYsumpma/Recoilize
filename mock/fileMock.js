// mock used for Jest config in package.json
module.exports = 'test-file-stub';