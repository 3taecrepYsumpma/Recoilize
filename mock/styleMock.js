// mock for jest CSS files, used in package.json for jest config
module.exports = {};