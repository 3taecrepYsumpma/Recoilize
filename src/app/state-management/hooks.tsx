// using custom hooks to eliminate need for typing below hooks in all typescript files
import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux';
import type {RootState, AppDispatch} from './index';

export const useAppDispatch = () => useDispatch<AppDispatch>();

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
