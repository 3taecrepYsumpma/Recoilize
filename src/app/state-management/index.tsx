import {configureStore /*, getDefaultMiddleware*/} from '@reduxjs/toolkit';
import throttleReducer from './slices/ThrottleSlice';
import zoomReducer from '../state-management/slices/ZoomSlice';
import snapshotReducer from '../state-management/slices/SnapshotSlice';
import atomNetworkReducer from '../state-management/slices/AtomNetworkSlice';
import filterReducer from '../state-management/slices/FilterSlice';
import selectedReducer from './slices/SelectedSlice';
import atomsAndSelectorsReducer from './slices/AtomsAndSelectorsSlice';

import {
  persistStore,
  persistReducer,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage/session';
import {combineReducers} from 'redux';

// const customizedPayloadAction = getDefaultMiddleware({
//   serializableCheck: false,
// });

const reducers = combineReducers({
  zoom: zoomReducer,
  throttle: throttleReducer,
  snapshot: snapshotReducer,
  atomNetwork: atomNetworkReducer,
  filter: filterReducer,
  selected: selectedReducer,
  atomsAndSelectors: atomsAndSelectorsReducer,
});

const persistConfig = {
  key: 'root',
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({serializableCheck: false}),
});

export let persistor = persistStore(store);
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

// exporting root reducer for use in testing
export const rootReducer = persistedReducer;
