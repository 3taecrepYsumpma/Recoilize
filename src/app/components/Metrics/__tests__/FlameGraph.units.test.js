import React from 'react';
import FlameGraph from '../FlameGraph';
import {fireEvent, generateStore, screen} from '../../../tests/testing';
import { snapshotHistoryMock } from '../../../../../mock/state-snapshot';

import {
  filteredCurSnapMock,
  componentAtomTreeMock,
} from '../../../../../mock/snapshot.js';
import {render, cleanup, queryByText} from '@testing-library/react';
import '@testing-library/dom';
import '@testing-library/jest-dom'
import { store } from '../../../state-management';

//Currently has 83% testing coverage up from 6%
/*            What should we be testing for the Flame Graph?
  - Tracks render time for each individual component PLUS its children at every level of the component tree
  - For our tests, the FlameGraph component needs the cleanedComponentTree (which is a state component), height, and width (these two we can create mock data for)
    - We need to test:
    - Rendering properly (x)
    - No initial values (should not be passing because graphs are rendering when we havent started the playground) 
    - Check that labels are properly being passed in  (x)
      - getting comopnents from the cleanedComponentAtomTree
*/

const mockHeight = 100
const mockWidth = 50

// define default store from state snapshot
const cleanedComponentAtomTreeData = store.getState().snapshot.cleanComponentAtomTree;
afterEach(cleanup);

describe('Flame graph displays correct information', () => {
  it('should render cleanedComponentAtomTree, width, and height', () => {
    const {asFragment} = render(
      <FlameGraph 
        cleanedComponentAtomTree={cleanedComponentAtomTreeData}
        width={mockWidth}
        height={mockHeight}
      />
    )
    expect(asFragment()).toMatchSnapshot();
  });
  //check that labels are being properly passed in 
  it('should contain cleanedComponentAtomTree', () => {
    // render flame graph
    render(
      <FlameGraph 
        cleanedComponentAtomTree={cleanedComponentAtomTreeData}
        width={mockWidth}
        height={mockHeight}
      />
    )
    const result = {
      children: [
        {
          actualDuration: 0.8450000004813774,
          children: [],
          name: 'PlaygroundStart',
          recoilNodes: [],
          tag: 0,
          treeBaseDuration: 0.5550000005314359,
          wasSuspended: false,
        },
      ],
      name: 'PlaygroundRender',
      recoilNodes: ['playStart'],
      tag: 0,
      actualDuration: 1.6750000004321919,
    };
    expect.objectContaining(result);

    // expect(screen.queryByText('Board')).toBeInTheDocument();
    //HAVING TROUBLE UNDERSTANDING HOW TO MOCK CLEANEDCOMPONENTATOMTREE, for now will just check for it, but would like to see that it includes 
    //everything inside that cleanedComponentAtomTree
    //ensure that the component has a cleanedComponentAtomTree property

  });

});

//is passing, but should not be because nothing should render when no data is passed in
describe('Flame graph initial rendering', () => {
  it('should not render when no values are passed in', () => {
    const {asFragment} = render(
      <FlameGraph 
        cleanedComponentAtomTree={componentAtomTreeMock}
        width={mockWidth}
        height={mockHeight}
      />
    )
    expect(asFragment()).toMatchSnapshot();
  });
});