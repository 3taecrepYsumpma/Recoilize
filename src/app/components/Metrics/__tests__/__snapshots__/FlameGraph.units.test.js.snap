// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flame graph displays correct information should render cleanedComponentAtomTree, width, and height 1`] = `
<DocumentFragment>
  <svg
    height="100"
    width="50"
  >
    <g
      class="vx-group"
      transform="translate(0, 0)"
    >
      <g
        transform="translate(1, 1)"
      >
        <rect
          fill-opacity="1"
          height="98"
          id="rect-0"
          width="48"
        />
        <clippath
          id="clip-0"
        >
          <use
            xlink:href="#rect-0"
          />
        </clippath>
        <text
          clip-path="url(#clip-0)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`Flame graph initial rendering should not render when no values are passed in 1`] = `
<DocumentFragment>
  <svg
    height="100"
    width="50"
  >
    <g
      class="vx-group"
      transform="translate(0, 0)"
    >
      <g
        transform="translate(1, 1)"
      >
        <rect
          fill-opacity="1"
          height="98"
          id="rect-0"
          width="8"
        />
        <clippath
          id="clip-0"
        >
          <use
            xlink:href="#rect-0"
          />
        </clippath>
        <text
          clip-path="url(#clip-0)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(10, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-1"
          width="9"
        />
        <clippath
          id="clip-1"
        >
          <use
            xlink:href="#rect-1"
          />
        </clippath>
        <text
          clip-path="url(#clip-1)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(10, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-2"
          width="9"
        />
        <clippath
          id="clip-2"
        >
          <use
            xlink:href="#rect-2"
          />
        </clippath>
        <text
          clip-path="url(#clip-2)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(10, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-3"
          width="9"
        />
        <clippath
          id="clip-3"
        >
          <use
            xlink:href="#rect-3"
          />
        </clippath>
        <text
          clip-path="url(#clip-3)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(20, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-4"
          width="8.999999999999996"
        />
        <clippath
          id="clip-4"
        >
          <use
            xlink:href="#rect-4"
          />
        </clippath>
        <text
          clip-path="url(#clip-4)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(20, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-5"
          width="8.999999999999996"
        />
        <clippath
          id="clip-5"
        >
          <use
            xlink:href="#rect-5"
          />
        </clippath>
        <text
          clip-path="url(#clip-5)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(20, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-6"
          width="8.999999999999996"
        />
        <clippath
          id="clip-6"
        >
          <use
            xlink:href="#rect-6"
          />
        </clippath>
        <text
          clip-path="url(#clip-6)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(20, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-7"
          width="8.999999999999996"
        />
        <clippath
          id="clip-7"
        >
          <use
            xlink:href="#rect-7"
          />
        </clippath>
        <text
          clip-path="url(#clip-7)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(30, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-8"
          width="9"
        />
        <clippath
          id="clip-8"
        >
          <use
            xlink:href="#rect-8"
          />
        </clippath>
        <text
          clip-path="url(#clip-8)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(30, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-9"
          width="9"
        />
        <clippath
          id="clip-9"
        >
          <use
            xlink:href="#rect-9"
          />
        </clippath>
        <text
          clip-path="url(#clip-9)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
      <g
        transform="translate(40, 1)"
      >
        <rect
          fill-opacity="1"
          height="0"
          id="rect-10"
          width="9"
        />
        <clippath
          id="clip-10"
        >
          <use
            xlink:href="#rect-10"
          />
        </clippath>
        <text
          clip-path="url(#clip-10)"
          style="font: sans-serif 10px 10px; font-weight: bold;"
          x="4"
          y="13"
        >
          <tspan
            style="font-size: 9px; fill-opacity: 0.8;"
          >
             NaN
          </tspan>
        </text>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;
